### ISSUE TITLE
resolveArrayConnection and null

### ISSUE THREAD

**Issue by jamiehodge** [hayes/giraphql#112]
The `resolveArrayConnection` helper is current resolving `T` to `Foo | null` without complaint. This will result in a runtime error because `node` is not nullable. As arrays can contain holes due to missing relations, I _think_ the best thing to do is to map null to null edges in the result.

This is my current workaround: 

```typescript
        const connection = resolveArrayConnection({ args }, foos);

        if (connection == null) {
          return null
        }

        // work around bug in resolveArrayConnection
        return {
          ...connection,
          edges: connection.edges.map(edge => edge?.node == null ? null : edge) || []
        }
```

**Comment by hayes** [comment]
I think pulling bulls up to the edge level makes sense, but I think there might be some other issues to fix here.  I'll try to look into this today

**Comment by hayes** [comment]
released a preview version that fixes this in #113

**Comment by jamiehodge** [comment]
Thank you!
